### PR TITLE
search: add `--deleted=` flag

### PR DESCRIPTION
**Pull request by FnControlOption**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

For example: `brew search --deleted="1 year ago" avian`

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
I think these should search deleted formulae by default I'm afraid. Are there any other options you can see to alter this behaviour? It could potentially be something disabled with an `env_config` if we thought it'd be more widely used?

**Comment by carlocab** [comment]
A switch might still be useful for searching for deleted formulae over a longer horizon than the last month.

File: Library/Homebrew/cmd/search.rb [hunk 1]
```diff
@@ -41,6 +41,8 @@ def search_args
              description: "Search online and locally for formulae."
       switch "--cask", "--casks",
              description: "Search online and locally for casks."
+      flag "--deleted=",
+           description: "Search for a formula deleted after a given date (default: `1 month ago`)."
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/search.rb:45]: How about adding this flag but defaulting to "forever"?

Comment (FnControlOption) [reply Library/Homebrew/cmd/search.rb:45]: When you say "default", do you mean "default without `--deleted`", or "default when `--deleted` is used without a date"?

Comment (carlocab) [reply Library/Homebrew/cmd/search.rb:45]: I think defaulting to forever when `--deleted` is used without a date would be good. Amending the existing message about searching for deleted formulae to suggest using `--deleted` to search a longer horizon might be helpful.

Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/search.rb:45]: > I think defaulting to forever when `--deleted` is used without a date would be good.

Yup, this 👍🏻

**Pull request closed without merging**